internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/util/JwtTokenUtil.java

```diff
@@ -0,0 +1,182 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.security.tools.scanmanager.webapp.util;
+
+import org.json.JSONObject;
+import org.springframework.core.io.ClassPathResource;
+import org.wso2.security.tools.scanmanager.webapp.exception.ScanManagerWebappException;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.util.HashMap;
+import javax.xml.bind.DatatypeConverter;
+
+/**
+ * Utility class for JWT Token.
+ */
+public class JwtTokenUtil {
+
+    private static JSONObject jwtPayload;
+
+    /**
+     * Validate the JWT token.
+     *
+     * @param jwtComponents array for JWT Token component (Header,Payload,Signature)
+     * @param config        JWT configuration to validate
+     * @return whether the JWT token is valid against the config
+     * @throws ScanManagerWebappException when an error occurs while validating JWT token
+     */
+    public static boolean validateJWT(String[] jwtComponents, HashMap config) throws ScanManagerWebappException {
+        jwtPayload = new JSONObject(getJWTDecode(jwtComponents[1]));
+        if (!validateMandatoryFields(jwtPayload)) {
+            String err = "Mandatory fields(Issuer, Subject, Expiration time or Audience) " +
+                    "are empty in the given JSON Web Token.";
+            throw new ScanManagerWebappException(err);
```

Comment:
These should be done inside the particular private validate methods. It is their responsibility.